native/src/main/java/io/ballerina/lib/avro/Utils.java

```diff
@@ -31,11 +36,34 @@ private Utils() {
 
     public static final String AVRO_SCHEMA = "avroSchema";
     public static final String ERROR_TYPE = "Error";
-    public static final String JSON_PROCESSING_ERROR = "JSON processing error";
-    public static final String DESERIALIZATION_ERROR = "Unable to deserialize the byte value";
+    public static final String SERIALIZATION_ERROR = "Avro serialization error";
+    public static final String DESERIALIZATION_ERROR = "Avro deserialization error";
+    public static final String STRING_TYPE = "BStringType";
+    public static final String ARRAY_TYPE = "BArrayType";
+    public static final String MAP_TYPE = "BMapType";
+    public static final String RECORD_TYPE = "BRecordType";
+    public static final String INTEGER_TYPE = "BIntegerType";
+    public static final String FLOAT_TYPE = "BFloatType";
 
     public static BError createError(String message, Throwable throwable) {
         BError cause = ErrorCreator.createError(throwable);
         return ErrorCreator.createError(getModule(), ERROR_TYPE, StringUtils.fromString(message), cause, null);
     }
+
+    public static Type getMutableType(IntersectionType intersectionType) {
+        for (Type type : intersectionType.getConstituentTypes()) {
+            Type referredType = TypeUtils.getImpliedType(type);
+            if (referredType instanceof UnionType) {
+                for (Type elementType : ((UnionType) referredType).getMemberTypes()) {
```

Comment:
There should be a `TypeUtils` method to get the element type from the union type. Shall we check whether that's the case? If so, let's use that API instead of manually iterating.

Reply:
Talked with the runtime team and tit looks like we'll need to manually iterate to extract the type from the union type